ext/snowcrash/src/ParametersParser.h

```diff
@@ -200,28 +197,49 @@ namespace snowcrash
     template <typename T>
     static void checkParametersEligibility(const MarkdownNodeIterator& node,
         const SectionParserData& pd,
-        Parameters& parameters,
+        const Parameters& parameters,
         const ParseResultRef<T>& out)
     {
 
-        for (ParameterIterator it = parameters.begin(); it != parameters.end(); ++it) {
-
-            if (!isValidUriTemplateParam(out.node.uriTemplate, it->name)) {
-
-                // WARN: parameter name not present
-                std::stringstream ss;
-                ss << "parameter '" << it->name << "' is not found within the URI template '" << out.node.uriTemplate
-                   << "'";
-
-                if (!out.node.name.empty()) {
-                    ss << " for '" << out.node.name << "' ";
+        using namespace apib::parser::uritemplate;
+        state::uritemplate result;
+
+        tao::pegtl::memory_input<> in(out.node.uriTemplate, "");
+        tao::pegtl::parse<match_grammar, action>(in, result);
+
+        // extract names of all variables
+        using Vars = std::vector<std::string>;
+        Vars variables;
+        auto collect = std::back_inserter(variables);
+
+        std::for_each(result.begin(), result.end(), 
+            [&collect](const state::part& part) {
+                if (const auto expression = mpark::get_if<state::expression>(&part)) {
+                  std::transform(expression->variables.begin(), expression->variables.end(),
+                      collect,
+                      [](const state::expression::variable_type& var){
+                        if (const auto valid = mpark::get_if<state::variable>(&var)) {
+                          return valid->name;
+                        } else if (const auto invalid = mpark::get_if<state::invalid>(&var)) {
+                          // FIXME: what to do with invalid variable names? Shouldn't we ignore them?
```

Comment:
I think that makes sense, we can filter them out.

Reply:
Okay, will remove check against invalid variables